launcher/java/JavaUtils.cpp

```diff
@@ -237,22 +231,25 @@ QList<JavaInstallPtr> JavaUtils::FindJavaFromRegistryKey(DWORD keyType, QString
                                       KEY_READ | KEY_WOW64_64KEY, &newKey) == ERROR_SUCCESS)
                     {
                         // Read the JavaHome value to find where Java is installed.
-                        value = new char[0];
+                        delete [] value;
+                        value = new TCHAR[0];
                         valueSz = 0;
                         if (RegQueryValueExW(newKey, keyJavaDir.toStdWString().c_str(), NULL, NULL, (BYTE *)value,
                                              &valueSz) == ERROR_MORE_DATA)
                         {
-                            value = new char[valueSz];
+                            value = new TCHAR[valueSz];
```

Comment:
This is where value should actually be declared.
```suggestion
                            // "+ 1" in case the integer floor division causes the result to be rounded down... Not sure if this is necessary because the Windows API docs don't mention it afaik, but better safe than sorry I guess.
                            WCHAR *value = new WCHAR[valueSz / sizeof(*value) + 1];
```